var/spack/repos/builtin/packages/py-nvidia-dali/package.py

```diff
@@ -21,36 +21,150 @@ class PyNvidiaDali(PythonPackage):
+        )
+        version(
+            "1.25.0-cuda110",
+            sha256="2eb94223ac980658606af6a56720ce963f4fd877c1291d08517f82ce435b4155",
+            expand=False,
+            preferred=False,
+            url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda110/nvidia_dali_cuda110-1.25.0-7922357-py3-none-manylinux2014_aarch64.whl",
+        )
+        version(
+            "1.24.0-cuda120",
+            sha256="2a7fab1d94b23edde1cee5b93918aca6b86417e3ffb4544adcb9961c73375014",
+            expand=False,
+            preferred=False,
+            url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda120/nvidia_dali_cuda120-1.24.0-7582307-py3-none-manylinux2014_aarch64.whl",
+        )
+        version(
+            "1.24.0-cuda110",
+            sha256="84711689dacc787dfd90bfc66da7ce4b1884a006b763109e9ecf0b07aefacbc2",
+            expand=False,
+            preferred=False,
+            url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda110/nvidia_dali_cuda110-1.24.0-7582302-py3-none-manylinux2014_aarch64.whl",
+        )
+        version(
+            "1.23.0-cuda120",
+            sha256="911d16b40c95b8cc700d3c96b40d3144953e7ffbb191ec22a75990c76cf638c3",
+            expand=False,
+            preferred=False,
+            url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda120/nvidia_dali_cuda120-1.23.0-7355174-py3-none-manylinux2014_aarch64.whl",
+        )
+        version(
+            "1.23.0-cuda110",
+            sha256="ca58f2990825d18736c872f48d3f5e5dbda8de136ab6339f1f9f6984d7b3dffe",
+            expand=False,
+            preferred=False,
+            url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda110/nvidia_dali_cuda110-1.23.0-7355173-py3-none-manylinux2014_aarch64.whl",
+        )
         version(
             "1.22.0-cuda120",
             sha256="5e496eebeba3bc1cddd18e081c8c45121283478931cbe9b64912d2394d0942ca",
             expand=False,
+            preferred=False,
             url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda120/nvidia_dali_cuda120-1.22.0-6971317-py3-none-manylinux2014_aarch64.whl",
         )
         version(
             "1.22.0-cuda110",
             sha256="0da47629fec01abf418fda0eeb393998820e40f6fae6b4c7d3e625aa4cdba6bd",
             expand=False,
-            preferred=True,
+            preferred=False,
             url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda110/nvidia_dali_cuda110-1.22.0-6988993-py3-none-manylinux2014_aarch64.whl",
         )
 
     depends_on("python@3.6:3.10", type=("build", "run"))
-    depends_on("cuda@12", when="@1.22.0-cuda120", type=("build", "run"))
-    depends_on("cuda@11", when="@1.22.0-cuda110", type=("build", "run"))
+    depends_on("cuda@12", when="@1.26.0-cuda120", type=("build", "run"))
+    depends_on("cuda@11", when="@1.26.0-cuda110", type=("build", "run"))
```

Comment:
I would do a for-loop over versions so that this applies to all versions. Unfortunately there isn't an easier way to do this.